diff --git a/core/src/main/java/org/jruby/ir/IREvalScript.java b/core/src/main/java/org/jruby/ir/IREvalScript.java
index 3fbfc9e7ca1..6f7a926cb65 100644
--- a/core/src/main/java/org/jruby/ir/IREvalScript.java
+++ b/core/src/main/java/org/jruby/ir/IREvalScript.java
@@ -8,6 +8,15 @@ import org.jruby.util.ByteList;
 public class IREvalScript extends IRClosure {
     private String fileName;
 
+    public int getNextClosureId() {
+        nextClosureIndex++;
+
+        return nextClosureIndex;
+    }
+
+    // Index values to guarantee we don't assign same internal index twice
+    private int nextClosureIndex;
+
     private static final ByteList EVAL_ = new ByteList(new byte[] {'E', 'V', 'A', 'L', '_'});
 
     public IREvalScript(IRManager manager, IRScope lexicalParent, String fileName,
